### PR TITLE
Initial add of fastq methods

### PR DESCRIPTION
**Pull request by tanaes**
This probably doesn't function right now, but I wanted to add to allow some initial review.

I am adding the FastQC and MultiQC commands as a method of the summary.py script. We will also want to configure validate.py to invoke these methods upon artifact creation.

### CODE REVIEW

**Comment by tanaes** [comment]
Hey @josenavas can you take a look at this so far and let me know if you see anything major that needs to be changed about the structure of the fastqc functionality? Thanks!

**Comment by josenavas** [comment]
Hi @tanaes thanks! I took a quick look. In general looks good, but Type plugins (as is this one) work a bit different from the other plugins. The main limitation is that they can't contain arbitrary Qiita commands. ATM there are only 2 possible commands: validate and generate HTML summary. I think this actually works in your benefit, since you only need to write a function that given the files that you need (passed by parameter) you run FastQC and MultiQC and return the output files.

This will work like a charm for generating those files automatically at upload time (i.e. that function will be called from the "validate" plugin function) so all files get added at the artifact creation time. However, I don't think it is going to be that easy to add this call to the Generate HTML summary unless we do some more changes in Qiita right now that I'm unsure on how to actually fix. We can chat at some point today why this is happening and potential solutions.

File: qtp_target_gene/fastqc.py [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+    bool, list of tup, str
+        The results of the job
+    """
+
+    # Get forward and (if exists) reverse seq filepaths
+    fwd_fps = [x for x, y in filepaths if y == 'raw_forward_seqs']
+    rev_fps = [x for x, y in filepaths if y == 'raw_reverse_seqs']
+
+    # Generate temp dir for exection of QC steps
+    out_dir = mkdtemp()
```
Comment (tanaes) [inline qtp_target_gene/fastqc.py:225]: @josenavas Currently this is creating a temporary directory for output, which worked fine when this was operating as a plugin. How do we do this as part of validate.py? I see that generate_html_summary takes out_dir as a required argument -- should we do the same here, and if so, how do we determine it?

**Comment by tanaes** [comment]
Ok so I can move this code over to validate and then integrate the fastqc
into the validate method for now?

**Comment by tanaes** [comment]
@josenavas Give me a shout when you're ready to start incorporating this.

**Pull request closed without merging**